homeassistant/components/device_tracker/icloud.py

```diff
@@ -335,22 +329,48 @@ def determine_interval(self, devicename, latitude, longitude, battery):
                  self._overridestates.get(devicename) == 'away')):
             return
 
+        zones = (self.hass.states.get(entity_id) for entity_id
+                 in sorted(self.hass.states.async_entity_ids('zone')))
```

Comment:
Don't call an async method from a sync context. Use `self.hass.states.entity_ids`.

Reply:
Thanks for the feedback.  I'll try to address it with a new pull request over the weekend.

Reply:
Opened a PR: https://github.com/home-assistant/home-assistant/pull/12890